contracts/exchangeIssuance/ExchangeIssuance.sol

```diff
@@ -0,0 +1,583 @@
+        address indexed _recipient,
+        address indexed _setToken,
+        address indexed _outputToken,
+        uint256 _amountSetToken,
+        uint256 _amountOut
+    );
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        address _uniFactory,
+        IUniswapV2Router02 _uniRouter, 
+        address _sushiFactory, 
+        IUniswapV2Router02 _sushiRouter, 
+        IBasicIssuanceModule _basicIssuanceModule
+    ) 
+        public
+    {
+        uniFactory = _uniFactory;
+        uniRouter = _uniRouter;
+
+        sushiFactory = _sushiFactory;
+        sushiRouter = _sushiRouter;
+
+        WETH = uniRouter.WETH();
+        basicIssuanceModule = _basicIssuanceModule;
+        IERC20(WETH).approve(address(uniRouter), PreciseUnitMath.maxUint256());
+        IERC20(WETH).approve(address(sushiRouter), PreciseUnitMath.maxUint256());
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * Runs all the necessary approval functions required before issuing
+     * or redeeming a set token. This function only need to be called before the first time
+     * this smart contract is used on any particular set token, or when a new token is added
+     * to the set token during a rebalance.
+     *
+     * @param _setToken    Address of the set token being initialized
+     */
+    function initApprovals(ISetToken _setToken) external {
+        ISetToken.Position[] memory positions = _setToken.getPositions();
+        for (uint256 i = 0; i < positions.length; i++) {
+            IERC20 token = IERC20(positions[i].component);
+            token.approve(address(uniRouter), MAX_UINT96);
+            token.approve(address(sushiRouter), MAX_UINT96);
+            token.approve(address(basicIssuanceModule), MAX_UINT96);
+        }
+    }
+
+    /**
+     * Issues an set token by swapping for the underlying tokens on Uniswap
+     * or Sushiswap.
+     *
+     * @param _setToken         Address of the set token being issued
+     * @param _amountInput      Amount of the input token / ether to spend
+     * @param _isInputETH       Set to true if the input token is Ether
+     * @param _inputToken       Address of input token. Ignored if _isInputETH is true
+     * @param _minSetReceive    Minimum amount of index to receive
+     */
+    function exchangeIssue(
+        ISetToken _setToken,
+        uint256 _amountInput,
+        bool _isInputETH,
+        IERC20 _inputToken,
+        uint256 _minSetReceive
+    ) 
+        external
+        payable
+        nonReentrant
+    {
+        _handleIssueInput(_isInputETH, _inputToken, _amountInput);
+
+        // get price of set token on uniswap
+        uint256 wethBalance = IERC20(WETH).balanceOf(address(this));
+        (uint256 tokenReserveA, uint256 tokenReserveB) = UniswapV2Library.getReserves(uniFactory, WETH, address(_setToken));
+        uint256 minSetTokenAmountOut = uniRouter.getAmountOut(wethBalance, tokenReserveA, tokenReserveB);
```

Comment:
Please include `sushiwap.getAmountOut()` here and set the `minSetTokenAmountOut` as `max(uniTokenAmountOut, sushiTokenAmountOut`.